### PR TITLE
Various cleanups to the ABI handling code (part 1)

### PR DESCRIPTION
**Pull request by bjorn3**
These are the relatively trivial parts split out of https://github.com/bytecodealliance/wasmtime/pull/8875. Each commit is a separate improvement.

### CODE REVIEW

**Comment by bjorn3** [comment]
https://github.com/bytecodealliance/wasmtime/pull/8903/commits/20850f8054c255a049f00a2f037d774695c7f449 is responsible for the crash. It can be reproduced using `cargo test --test wast -- Cranelift/pooling/tests/misc_testsuite/custom-page-sizes/custom-page-sizes.wast`.

**Comment by bjorn3** [comment]
Never mind. It actually crashes on the main branch too.

**Comment by bjorn3** [comment]
Gentle ping

File: cranelift/codegen/src/machinst/abi.rs [hunk 1]
```diff
@@ -2297,47 +2297,57 @@ impl<M: ABIMachineSpec> CallSite<M> {
...
-        into_regs: ValueRegs<Writable<Reg>>,
-    ) -> SmallInstVec<M::I> {
+    ) -> (SmallInstVec<M::I>, ValueRegs<Reg>) {
         let mut insts = smallvec![];
-        match &ctx.sigs().rets(self.sig)[idx] {
-            &ABIArg::Slots { ref slots, .. } => {
-                assert_eq!(into_regs.len(), slots.len());
-                for (slot, into_reg) in slots.iter().zip(into_regs.regs().iter()) {
+        let mut into_regs: SmallVec<[Reg; 2]> = smallvec![];
+        let arg = ctx.sigs().rets(self.sig)[idx].clone();
```
Comment (cfallin) [inline cranelift/codegen/src/machinst/abi.rs:2310]: Can we call this `ret` rather than `arg` for consistency (with `rets`)?

**Review by cfallin — APPROVED** [review]
This looks great, thanks! I like the functional style (returning result registers) for `gen_retval`, that simplified things greatly. One naming request below otherwise good to merge.

File: cranelift/codegen/src/machinst/abi.rs (hunk 1 above)
Comment (bjorn3) [reply cranelift/codegen/src/machinst/abi.rs]: Done

**Pull request merged**